src/main/cpp/src/timezones.cu

```diff
@@ -163,4 +342,50 @@ std::unique_ptr<column> convert_utc_timestamp_to_timezone(column_view const& inp
   return convert_timestamp(input, transitions, tz_index, false, stream, mr);
 }
 
+std::unique_ptr<column> time_add(column_view const& input,
+                                 scalar_i64 const& duration,
+                                 table_view const& transitions,
+                                 size_type tz_index,
+                                 rmm::cuda_stream_view stream,
+                                 rmm::mr::device_memory_resource* mr)
+{
+  auto const type = input.type().id();
+
+  switch (type) {
+    case cudf::type_id::TIMESTAMP_SECONDS:
+      return time_add_with_tz<cudf::timestamp_s>(
+        input, duration, transitions, tz_index, stream, mr);
+    case cudf::type_id::TIMESTAMP_MILLISECONDS:
+      return time_add_with_tz<cudf::timestamp_ms>(
+        input, duration, transitions, tz_index, stream, mr);
+    case cudf::type_id::TIMESTAMP_MICROSECONDS:
+      return time_add_with_tz<cudf::timestamp_us>(
+        input, duration, transitions, tz_index, stream, mr);
+    default: CUDF_FAIL("Unsupported timestamp unit for timezone conversion");
+  }
+}
+
+std::unique_ptr<column> time_add(column_view const& input,
+                                 column_view const& duration,
```

Comment:
What is the supported duration type(s)? CUDF supports multiple ones.

https://github.com/rapidsai/cudf/blob/726a7f30757d1a06d74d86bb82cf311cb159f7fd/cpp/include/cudf/types.hpp#L205-L209

Spark also supports multiple ones.

https://github.com/apache/spark/blob/ebd2b78f87fa6086c41d5e6bcade5efeefac75d0/common/unsafe/src/main/java/org/apache/spark/unsafe/types/CalendarInterval.java#L50-L52

https://github.com/apache/spark/blob/ebd2b78f87fa6086c41d5e6bcade5efeefac75d0/sql/api/src/main/scala/org/apache/spark/sql/types/DayTimeIntervalType.scala#L27-L30

https://github.com/apache/spark/blob/ebd2b78f87fa6086c41d5e6bcade5efeefac75d0/sql/api/src/main/scala/org/apache/spark/sql/types/YearMonthIntervalType.scala#L27-L28

The code appears to assume that the duration must be in microseconds. That might be fine in the short term, but I don't see anywhere that we check for this condition and enforce it. We just assume that the input is 64 bits and treat it like it was in microseconds.